### PR TITLE
Fixed #25142 -- Refactor access mixins to allow customisation

### PR DESCRIPTION
**Pull request by dfunckt**
Related ticket: [#25142](https://code.djangoproject.com/ticket/25142)

### CODE REVIEW

File: django/contrib/auth/mixins.py [hunk 1]
```diff
@@ -79,9 +79,12 @@ def get_permission_required(self):
             perms = self.permission_required
         return perms
 
+    def check_for_permission(self, user, perms):
```
Comment (MarkusH) [inline django/contrib/auth/mixins.py]: Can you drop the user argument, please. You already gave all information in `self....`.

**Comment by timgraham** [comment]
buildbot, add to whitelist.

**Comment by timgraham** [comment]
merged in 29465d438e00660221e1e0ac5e2929a719c56113, thanks!

**Comment by dfunckt** [comment]
Thanks Tim, Markus!

**Comment by MarkusH** [comment]
You're welcome, @dfunckt :smile:  Thanks for the patch :sparkles:

**Pull request closed without merging**